front50-sql/src/main/kotlin/com/netflix/spinnaker/front50/model/SqlStorageService.kt

```diff
@@ -164,20 +167,76 @@ class SqlStorageService(
     return objects
   }
 
+  override fun <T : Timestamped> loadObjectsNewerThan(
+    objectType: ObjectType,
+    lastModifiedThreshold: Long
+  ):
+    Map<String, List<T>> {
+
+    log.debug("Fetching {} objects with last_modified_at value greater than {}",
+      objectType.name,
+      lastModifiedThreshold)
+
+    val deletedKey = "deleted"
+    val notDeletedKey = "not_deleted"
+
+    val resultMap = mapOf<String, MutableList<T>>(
+      deletedKey to mutableListOf(),
+      notDeletedKey to mutableListOf()
+    )
+
+    val tableSupportsSoftDeletes = definitionsByType[objectType]!!.supportsHistory
+
+    val fieldsToFetch = mutableListOf<SelectFieldOrAsterisk>(field("body", String::class.java))
```

Comment:
Sort of a nitpick, but I've recently started using [this `SelectQuery` API](https://www.jooq.org/javadoc/3.14.x/org.jooq/org/jooq/SelectQuery.html) for assembling dynamic queries like this. This might simplify things for you. See also [`InsertQuery`](https://www.jooq.org/javadoc/3.14.x/org.jooq/org/jooq/InsertQuery.html) which also abstracts away some MySQL/Postgres differences in the plain DSL.

Reply:
These sound like good improvements.  Any chance we can leave them for a followup PR?  We've been running this code as is for a long time and I'd love to avoid another round of testing.